File: Test12/step_definitions/Shopping Cart Management_2.java
package com.cucumber.steps;

import com.framework.cucumber.TestHarness;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import com.page_objects.CartPage;

public class CartManagementSteps extends CartPage {

    private TestHarness testHarness = new TestHarness();

    @Given("a user is on the product page")
    public void aUserIsOnTheProductPage() {
        navigateToProductPage();
    }

    @When("the user adds a product to the shopping cart")
    public void theUserAddsAProductToTheShoppingCart() {
        addProductToCart();
    }

    @Then("the product should be added to the shopping cart")
    public void theProductShouldBeAddedToTheShoppingCart() {
        verifyProductInCart();
    }

    @Then("the user should receive a notification of successful addition")
    public void theUserShouldReceiveANotificationOfSuccessfulAddition() {
        verifyAdditionNotification();
    }

    @Then("the system should log the add operation for auditing")
    public void theSystemShouldLogTheAddOperationForAuditing() {
        verifyAddOperationLogged();
    }

    @Given("a product is present in the shopping cart")
    public void aProductIsPresentInTheShoppingCart() {
        ensureProductInCart();
    }

    @When("the user removes the product from the shopping cart")
    public void theUserRemovesTheProductFromTheShoppingCart() {
        removeProductFromCart();
    }

    @Then("the product should be removed from the shopping cart")
    public void theProductShouldBeRemovedFromTheShoppingCart() {
        verifyProductNotInCart();
    }

    @Then("the user should receive a notification of successful removal")
    public void theUserShouldReceiveANotificationOfSuccessfulRemoval() {
        verifyRemovalNotification();
    }

    @Then("the system should log the remove operation for auditing")
    public void theSystemShouldLogTheRemoveOperationForAuditing() {
        verifyRemoveOperationLogged();
    }

    @When("the user modifies the quantity of the product")
    public void theUserModifiesTheQuantityOfTheProduct() {
        modifyProductQuantity();
    }

    @Then("the quantity should be updated immediately in the shopping cart")
    public void theQuantityShouldBeUpdatedImmediatelyInTheShoppingCart() {
        verifyQuantityUpdated();
    }

    @Then("the system should update the subtotal based on the new quantity")
    public void theSystemShouldUpdateTheSubtotalBasedOnTheNewQuantity() {
        verifySubtotalUpdated();
    }

    @Then("the user should receive a notification of successful modification")
    public void theUserShouldReceiveANotificationOfSuccessfulModification() {
        verifyModificationNotification();
    }

    @Then("the system should log the modify operation for auditing")
    public void theSystemShouldLogTheModifyOperationForAuditing() {
        verifyModifyOperationLogged();
    }

    @When("the user attempts to set a negative quantity for the product")
    public void theUserAttemptsToSetANegativeQuantityForTheProduct() {
        attemptNegativeQuantity();
    }

    @Then("the system should prevent the negative quantity")
    public void theSystemShouldPreventTheNegativeQuantity() {
        verifyNegativeQuantityPrevention();
    }

    @Then("the user should receive an error message that is user-friendly")
    public void theUserShouldReceiveAnErrorMessageThatIsUserFriendly() {
        verifyUserFriendlyErrorMessage();
    }

    @Given("a user is on a desktop or mobile platform")
    public void aUserIsOnADesktopOrMobilePlatform() {
        verifyPlatform();
    }

    @When("the user performs cart operations")
    public void theUserPerformsCartOperations() {
        performCartOperations();
    }

    @Then("the operations should be accessible and function correctly on both platforms")
    public void theOperationsShouldBeAccessibleAndFunctionCorrectlyOnBothPlatforms() {
        verifyOperationsOnPlatforms();
    }

    @Given("a user is performing cart operations")
    public void aUserIsPerformingCartOperations() {
        initiateCartOperations();
    }

    @When("an error occurs")
    public void anErrorOccurs() {
        simulateError();
    }

    @Then("the system should handle the error gracefully")
    public void theSystemShouldHandleTheErrorGracefully() {
        verifyErrorHandling();
    }

    @Then("provide a user-friendly message to the user")
    public void provideAUserFriendlyMessageToTheUser() {
        verifyUserFriendlyErrorMessage();
    }
}

File: Test12/step_definitions/Shopping Cart Management_2.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import ShoppingCartPage from "../pages/shoppingCartPage";

Given("a user is on the product page", () => {
 ShoppingCartPage.visitProductPage();
});

When("the user adds a product to the shopping cart", () => {
 ShoppingCartPage.addProductToCart();
});

Then("the product should be added to the shopping cart", () => {
 ShoppingCartPage.verifyProductAdded();
});

Then("the user should receive a notification of successful addition", () => {
 ShoppingCartPage.verifyAdditionNotification();
});

Then("the system should log the add operation for auditing", () => {
 ShoppingCartPage.verifyAddOperationLogged();
});

Given("a product is present in the shopping cart", () => {
 ShoppingCartPage.verifyProductInCart();
});

When("the user removes the product from the shopping cart", () => {
 ShoppingCartPage.removeProductFromCart();
});

Then("the product should be removed from the shopping cart", () => {
 ShoppingCartPage.verifyProductRemoved();
});

Then("the user should receive a notification of successful removal", () => {
 ShoppingCartPage.verifyRemovalNotification();
});

Then("the system should log the remove operation for auditing", () => {
 ShoppingCartPage.verifyRemoveOperationLogged();
});

When("the user modifies the quantity of the product", () => {
 ShoppingCartPage.modifyProductQuantity();
});

Then("the quantity should be updated immediately in the shopping cart", () => {
 ShoppingCartPage.verifyQuantityUpdated();
});

Then("the system should update the subtotal based on the new quantity", () => {
 ShoppingCartPage.verifySubtotalUpdated();
});

Then("the user should receive a notification of successful modification", () => {
 ShoppingCartPage.verifyModificationNotification();
});

Then("the system should log the modify operation for auditing", () => {
 ShoppingCartPage.verifyModifyOperationLogged();
});

When("the user attempts to set a negative quantity for the product", () => {
 ShoppingCartPage.setNegativeQuantity();
});

Then("the system should prevent the negative quantity", () => {
 ShoppingCartPage.verifyNegativeQuantityPrevented();
});

Then("the user should receive an error message that is user-friendly", () => {
 ShoppingCartPage.verifyErrorMessage();
});

Given("a user is on a desktop or mobile platform", () => {
 ShoppingCartPage.verifyPlatformAccess();
});

When("the user performs cart operations", () => {
 ShoppingCartPage.performCartOperations();
});

Then("the operations should be accessible and function correctly on both platforms", () => {
 ShoppingCartPage.verifyOperationsFunctionality();
});

Given("a user is performing cart operations", () => {
 ShoppingCartPage.performCartOperations();
});

When("an error occurs", () => {
 ShoppingCartPage.simulateError();
});

Then("the system should handle the error gracefully", () => {
 ShoppingCartPage.verifyErrorHandling();
});

Then("provide a user-friendly message to the user", () => {
 ShoppingCartPage.verifyUserFriendlyMessage();
});